### PR TITLE
QPPCT-624: Re-evaluate application logging

### PR DESCRIPTION
**Pull request by halprin**
### Information
- JIRA story QPPCT-624.

### Changes proposed in this PR:
- Logs from Spring are logging again.
- Adds some additional logging in the ReST API.
- Silences the unused Spring default password log.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public String convert(ILoggingEvent event) {
 
 		try {
 			hashPart = getHashPart();
-		} catch (IOException | ServletException e) {
+		} catch (Exception e) {
```
Comment (sfradkin) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: Were there other exceptions that need to be caught and logged here?  I'd rather not catch `Exception` and be more specific.

File: rest-api/src/main/resources/logback.xml [hunk 2]
```diff
@@ -17,7 +17,7 @@
 			</providers>
 		</encoder>
 	</appender>
-	<root name="gov.cms" level="INFO">
+	<root level="INFO">
```
Comment (sfradkin) [inline rest-api/src/main/resources/logback.xml]: Do we want everything logging at INFO?  Will that output too much framework logging?

File: rest-api/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,3 +1,4 @@
 spring.main.banner-mode=off
 spring.http.multipart.max-file-size=-1
 spring.http.multipart.max-request-size=-1
+logging.level.org.springframework.boot.autoconfigure.security.AuthenticationManagerConfiguration=OFF
```
Comment (halprin) [inline rest-api/src/main/resources/application.properties]: Once @clydet's change to stop the password from logging goes in, we will remove this.

File: rest-api/src/main/resources/logback.xml (hunk 2 above)
Comment (halprin) [reply rest-api/src/main/resources/logback.xml]: _I_ don't think it's too noisy for one reason: all the framework logging that does occur is at start-up.  After that, there's nothing except our own application's.

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java (hunk 1 above)
Comment (omnivion) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: agreed

Comment (halprin) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: Good catch.  Only thing is that we cannot log here (even if we do it at a `INFO` or higher level).  Logback is smart enough to know that this piece of code is executing while constructing a log statement and therefore squelches any logging.  We also don't want to manually log anything here because it will become noisy, ruin the JSON format of our logs, and be hard to follow.

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java [hunk 4]
```diff
@@ -29,8 +26,8 @@ public String convert(ILoggingEvent event) {
 
 		try {
 			hashPart = getHashPart();
-		} catch (IOException | ServletException e) {
-			DEV_LOG.trace("No part to associate with log output.", e);
+		} catch (IOException | ServletException | NullPointerException e) {
```
Comment (FlexionSonarQube) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: MAJOR Avoid catching NullPointerException. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1696)
MAJOR Either log or rethrow this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java (hunk 4 above)
Comment (halprin) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: Shush @FlexionSonarQube.

**Comment by saquino0827** [comment]
@halprin should we add some logging in `PathCorrelator` due to the mapping issue found from the nava submission validation api bug? If so I can add that.

**Comment by halprin** [comment]
@saquino0827 I've been thinking about that too...  Yes, I think so.  I had an specific thought about how to do this, so want to pair tomorrow on this?

File: rest-api/src/main/resources/logback.xml (hunk 2 above)
Comment (omnivion) [reply rest-api/src/main/resources/logback.xml]: agreed with @halprin

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java (hunk 4 above)
Comment (sfradkin) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: @FlexionSonarQube has a point here.  NPE is a runtime exception that we generally don't want to catch.  The other two are checked exceptions.

Comment (halprin) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: Hmmm, we could just check for `null` around https://github.com/CMSgov/qpp-conversion-tool/pull/598/files#diff-5eed8e91cf01031595a455f39992335fR60 instead.  One of those chained method calls returns `null` which subsequently throws a `NullPointerException` in the early stages of the application's start-up.

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java [hunk 5]
```diff
@@ -30,7 +27,7 @@ public String convert(ILoggingEvent event) {
 		try {
 			hashPart = getHashPart();
 		} catch (IOException | ServletException e) {
```
Comment (FlexionSonarQube) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: MAJOR Either log or rethrow this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java (hunk 5 above)
Comment (halprin) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/logging/AttachmentHashPartConverter.java]: Nope, bad idea.  This is in a logging support class and doing either of this will destroy the output of our logs.  Go home, you're drunk.  😝
